private/bufpkg/bufprotosource/bufprotosource.go

```diff
@@ -358,6 +358,11 @@ type Enum interface {
 
 	// Will return nil if this is a top-level Enum
 	Parent() Message
+
+	// AsDescriptor returns a [protoreflect.Descriptor] that
```

Comment:
On face value, I don't like that we are doing this - `bufprotosource` exposing `protoreflect` types effectively just means we're exposing two versions of a substantially-similar API to the `bufcheck/...` packages. It just feels like it may devolve in the future - I'd prefer we either (1) get rid of `bufprotosource` entirely (we're not staffing this refactor) or (2) continue to lean into `bufprotosource` (which I think is actually significantly cleaner for the linter/breaking change detector use case anyways, having just attempted to write a lint plugin against `protoreflect` in https://github.com/bufbuild/buf/tree/bufmod-lint-plugins).

I may discover why this is necessary in this commit or the next one, however.

Reply:
Talked offline, fine with this - would be nice SomedayTM to have all types inherit from the protoreflect types